src/main/java/de/tum/in/www1/artemis/web/rest/FileResource.java

```diff
@@ -53,38 +60,51 @@
+
     public FileResource(FileService fileService, ResourceLoader resourceLoader, UserService userService, AuthorizationCheckService authCheckService,
-            LectureRepository lectureRepository, TokenProvider tokenProvider) {
+            LectureRepository lectureRepository, TokenProvider tokenProvider, FileUploadSubmissionRepository fileUploadSubmissionRepository,
+            FileUploadExerciseRepository fileUploadExerciseRepository) {
         this.fileService = fileService;
         this.resourceLoader = resourceLoader;
         this.userService = userService;
         this.authCheckService = authCheckService;
         this.lectureRepository = lectureRepository;
         this.tokenProvider = tokenProvider;
+        this.fileUploadSubmissionRepository = fileUploadSubmissionRepository;
+        this.fileUploadExerciseRepository = fileUploadExerciseRepository;
     }
 
     /**
      * POST /fileUpload : Upload a new file.
      *
      * @param file The file to save
      * @param keepFileName specifies if original file name should be kept
+     * @param isSubmission specifies if file is submitted by a student
      * @return The path of the file
      * @throws URISyntaxException if response path can't be converted into URI
      */
     @PostMapping("/fileUpload")
-    @PreAuthorize("hasAnyRole('ADMIN', 'INSTRUCTOR', 'TA')")
-    public ResponseEntity<String> saveFile(@RequestParam(value = "file") MultipartFile file, @RequestParam("keepFileName") Boolean keepFileName) throws URISyntaxException {
+    @PreAuthorize("hasAnyRole('ADMIN', 'INSTRUCTOR', 'TA', 'USER')")
+    public ResponseEntity<String> saveFile(@RequestParam(value = "file") MultipartFile file, @RequestParam("keepFileName") Boolean keepFileName,
+            @RequestParam(value = "isSubmission", required = false, defaultValue = "false") Boolean isSubmission) throws URISyntaxException {
         log.debug("REST request to upload file : {}", file.getOriginalFilename());
 
         // NOTE: Maximum file size is set in resources/config/application.yml
         // Currently set to 10 MB
 
+        if (isSubmission && file.getSize() > 2_000_000) {
+            // NOTE: Maximum file size for submission is 2 MB
+            return ResponseEntity.status(413).body("Maximum file size for submission is 2 MB!");
+        }
         // check for file type
         String fileExtension = FilenameUtils.getExtension(file.getOriginalFilename());
-        if (!fileExtension.equalsIgnoreCase("png") && !fileExtension.equalsIgnoreCase("jpg") && !fileExtension.equalsIgnoreCase("jpeg") && !fileExtension.equalsIgnoreCase("svg")
-                && !fileExtension.equalsIgnoreCase("pdf") && !fileExtension.equalsIgnoreCase("zip")) {
+        List<String> allowedFileExtensions = Arrays.asList("png", "jpg", "zip", "pdf", "svg", "jpeg");
+        if (!isSubmission && !allowedFileExtensions.stream().anyMatch(fileExtension::equalsIgnoreCase)) {
```

Comment:
You do not check if the file submitted by a student has the correct file ending as specified in the corresponding FileUploadExercise.filePattern. 

However, this is important, otherwise students could upload any file using this endpoint!

Reply:
Added a check on the server in the new endpoint